mne_bids/write.py

```diff
@@ -790,12 +790,13 @@ def write_raw_bids(raw, bids_basename, output_path, events_data=None,
         raise ValueError('raw.filenames is missing. Please set raw.filenames'
                          'as a list with the full path of original raw file.')
 
+    raw_fname = raw.filenames[0]
+
     if raw.preload is not False:
-        raise ValueError('The data should not be preloaded.')
+        raw = Raw(raw_fname, preload=False)
```

Comment:
This could lead to surprising outcomes, which is why we prefer to raise an error. Imagine if the user has modified the `raw` data after loading it with `preload=True` and you load the data again here using `Raw`. What you write to disk then does not reflect what the user expects to be written.